services/web/server/src/simcore_service_webserver/login/_auth_handlers.py

```diff
@@ -214,11 +263,14 @@ async def login_2fa(request: web.Request):
     login_2fa_ = await parse_request_body_as(LoginTwoFactorAuthBody, request)
 
     # validates code
-    if login_2fa_.code.get_secret_value() != await get_2fa_code(
-        request.app, login_2fa_.email
-    ):
+    _redis_2fa_code = await get_2fa_code(request.app, login_2fa_.email)
```

Comment:
Aha, so you want different messages in these two cases.

MINOR: i would rename as `expected_2fa_code` instead of `_redis_2fa_code`